src/network/networkd-dhcp4.c

```diff
@@ -1105,20 +1105,37 @@ static int dhcp4_handler(sd_dhcp_client *client, int event, void *userdata) {
                                 return r;
                         }
 
+                        if (link->ipv4ll) {
+                                r = sd_ipv4ll_stop(link->ipv4ll);
```

Comment:
`sd_ipv4ll_stop` immediately returns if the argument is NULL, so this `if` check in unnecessary.

A question though: does this result in the address being immediately dropped or just the address not being preferred anymore? If we drop the address immediately, existing connections will be broken.